File: tests/verify/managed/fixtures/clean/clean.dart
// clean.dart — clean Dart fixture: ZERO findings expected.

@weftHot
void onSignal(double x, Float64List out, Float64List scratch) {
  var acc = 0.0;
  for (var i = 0; i < 64; i++) {
    scratch[i] = x * i;
    acc += scratch[i];
  }
  out[0] = acc;
  if (acc > 1e9) {
    out[1] = -1.0;
    return;
  }
}

File: tests/verify/managed/fixtures/clean/clean.cpp
#include <cstddef>

// clean.cpp — clean C++ fixture: ZERO findings expected.

__attribute__((weft_hot))
void apply_tick(const Order &o, Stats *stats) {
    double acc = 0.0;
    for (int i = 0; i < o.count; i++) {
        acc += o.legs[i].px;
    }
    stats->sum = acc;
    stats->count = o.count;
}
